core/GlobalState.h

```diff
@@ -532,6 +548,10 @@ class GlobalState final {
         return this->files->span();
     }
 
+    std::shared_ptr<FileTable> getFileTable() const {
+        return this->files;
+    }
```

Comment:
w.r.t. the above, I think the fact that we need this function, to expose something that was not previously exposed, could indicate that storing this vector here in the FileTable is out of place?